src/main/java/org/folio/innreach/batch/contribution/service/InstanceLoader.java

```diff
@@ -29,16 +29,18 @@ public Instance load(InstanceIterationEvent event) {
       return null;
     }
 
-    // if the returned instance is null it is assumed that processing of the event should not continue
-    var instance = inventoryService.getInstance(event.getInstanceId());
+    var instanceId = event.getInstanceId();
+
+    var instance = inventoryService.getInstance(instanceId);
 
+    // if the returned instance is null it is assumed that processing of the event should not continue
     if (instance == null) {
-      log.info("No instance found by id {}", event.getInstanceId());
+      log.info("No instance found by id {}", instanceId);
       return null;
     }
 
-    if (!isMARCRecord(instance)) {
-      log.info("Source {} is not supported", instance.getSource());
+    if (!validationService.isEligibleForContribution(getCentralServerId(), instance)) {
+      log.info("Instance {} is not eligible for contribution", instanceId);
```

Comment:
Do we want to record what is exactly wrong with the instance so that it cannot be contributed?

Reply:
Do you mean saving the reason to DB?

Reply:
I mean at least adding this information to log.. Not sure what kind of output saved to DB. If we store some information per record, it make sense to save the reason, otherwise a log entry should be enough